calls/v1/observationunits/TripalWebServiceBrapiV1Observationunits.inc

```diff
@@ -146,6 +151,38 @@ class TripalWebServiceBrapiV1Observationunits extends TripalWebServiceDatabaseCa
       $args[':germplasmDbId'] = $call_asset['parameter']['germplasmDbId'];
     }
 
+    // observationVariableDbId.
+    if (isset($call_asset['parameter']['observationVariableDbId'])
+        && !empty($call_asset['parameter']['observationVariableDbId'])) {
+
+      $filter[] = 'obs.observationvariabledbid = :observationVariableDbId';
+      $args[':observationVariableDbId'] = $call_asset['parameter']['observationVariableDbId'];
+    }
+
+    // studyDbId.
+    if (isset($call_asset['parameter']['studyDbId'])
+        && !empty($call_asset['parameter']['studyDbId'])) {
+
+      $filter[] = 'prg.project_id = :studyDbId';
+      $args[':studyDbId'] = $call_asset['parameter']['studyDbId'];
+    }
+
+    // locationDbId.
+    if (isset($call_asset['parameter']['locationDbId'])
+        && !empty($call_asset['parameter']['locationDbId'])) {
+
+      $filter[] = "CONCAT(year.value, '-', loc.value, '-', rep.value, '-', germ.name, '-', unit.name) = :locationDbId";
+      $args[':locationDbId'] = $call_asset['parameter']['locationDbId'];
+    }
+
+    // trialDbId.
+    if (isset($call_asset['parameter']['trialDbId'])
+        && !empty($call_asset['parameter']['trialDbId'])) {
+
+      $filter[] = "CONCAT(year.value, '-', loc.value, '-', rep.value, '-', germ.name, '-', unit.name) = :locationDbId";
```

Comment:
This appears to be a copy/paste from location and doesn't actually filter for trialDbId.

Reply:
Good catch!